pact/provider/core/src/main/java/org/arquillian/algeron/pact/provider/core/StateTypeConverter.java

```diff
@@ -0,0 +1,149 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+public class StateTypeConverter {
```

Comment:
Can we also bring https://github.com/arquillian/arquillian-extension-persistence/blob/master/core/src/test/java/org/jboss/arquillian/persistence/core/configuration/ConfigurationTypeConverterTest.java ?

Out of curiosity - why you don't support primitives?